src/composables/pools/usePoolMigration.ts

```diff
@@ -0,0 +1,243 @@
+
+  /**
+   * STATE
+   */
+  const signature = ref('');
+
+  const actions = computed(() => {
+    const arr: TransactionActionInfo[] = [];
+
+    if (isStakedMigrationEnabled) {
+      arr.push({
+        label: t('migratePool.previewModal.actions.staked.title'),
+        loadingLabel: t('migratePool.previewModal.actions.loading'),
+        confirmingLabel: t('migratePool.confirming'),
+        action: approveMigration.bind(
+          null,
+          true,
+          scaleNum(stakedBptBalance, fromPool.onchain?.decimals)
+        ),
+        stepTooltip: t('migratePool.previewModal.actions.migrationStep'),
+        isSignAction: false,
+      });
+    }
+
+    if (isUnstakedMigrationEnabled) {
+      const data = {
+        label: t('migratePool.previewModal.actions.unstaked.title'),
+        loadingLabel: t('migratePool.previewModal.actions.loading'),
+        confirmingLabel: t('migratePool.confirming'),
+        action: approveMigration.bind(
+          null,
+          false,
+          scaleNum(unstakedBptBalance, fromPool.onchain?.decimals)
+        ),
+        stepTooltip: t('migratePool.previewModal.actions.migrationStep'),
+        isSignAction: false,
+      };
+
+      // the biggest one is migrated first
+      Number(unstakedAmount) > Number(stakedAmount)
+        ? arr.unshift(data)
+        : arr.push(data);
+    }
+
+    if (!relayerApproval.value) {
+      arr.unshift({
+        label: t('migratePool.approve'),
+        loadingLabel: t('checkWallet'),
+        confirmingLabel: t('migratePool.approving'),
+        stepTooltip: t('approveBatchRelayerTooltip'),
+        action: getUserSignature as () => Promise<any>,
+        isSignAction: true,
+      });
+    }
+    return arr;
+  });
+
+  const migrationType = computed(() => {
+    return POOL_MIGRATIONS.find(
+      migration => migration.fromPoolId === fromPool.id
+    );
+  });
+
+  /**
+   * METHODS
+   */
+  async function getUserSignature(): Promise<string> {
+    try {
+      const deadline = new Date().getTime() + HALF_HOUR;
+      const vaultInstance = balancerContractsService.vault?.instance;
+      const nonce = await vaultInstance.getNextNonce(account.value);
+      const relayerAddress = configService.network.addresses.batchRelayer;
+      const calldata = vaultInstance.interface.encodeFunctionData(
+        'setRelayerApproval',
+        [account.value, relayerAddress, true]
+      );
+
+      const _signature =
+        await RelayerAuthorization.signSetRelayerApprovalAuthorization(
+          vaultInstance,
+          getSigner(),
+          relayerAddress,
+          calldata,
+          deadline,
+          nonce
+        );
+
+      const encodedSign = RelayerAuthorization.encodeCalldataAuthorization(
+        '0x',
+        deadline,
+        _signature
+      );
+
+      signature.value = encodedSign;
+      return signature.value;
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  }
+
+  async function approveMigration(
```

Comment:
I think this function is too big. It should be split up if possible into smaller more concise functions.